File: Backend/AstroGame.Shared/Models/Players/Species.cs
using AstroGame.Shared.Enums;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace AstroGame.Shared.Models.Players
{
    public class Species
    {
        public Guid Id { get; set; }
        public SpeciesType SpeciesType { get; set; }
        public string AssetName { get; set; }

        [JsonIgnore] public List<PlayerSpecies> PlayerSpecies { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/players/species.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/enums/species_type.dart';

import 'package:json_annotation/json_annotation.dart';

part 'species.g.dart';

@GuidConverter()
@JsonSerializable()
class Species {
  late Guid id;
  late SpeciesType speciesType;
  late String assetName;

  Species();

  factory Species.fromJson(Map<String, dynamic> json) =>
      _$SpeciesFromJson(json);
  Map<String, dynamic> toJson() => _$SpeciesToJson(this);
}
